x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/date/DateParse.java

```diff
@@ -63,17 +81,80 @@ public DateParse(
             name = "dateString",
             type = { "keyword", "text" },
             description = "Date expression as a string. If `null` or an empty string, the function returns `null`."
-        ) Expression second
+        ) Expression second,
+        @MapParam(
+            name = "options",
+            params = {
+                @MapParam.MapParamEntry(
+                    name = TIME_ZONE_PARAM_NAME,
+                    type = "keyword",
+                    valueHint = { "standard" },
+                    description = "Coordinated Universal Time (UTC) offset or IANA time zone used to convert date values in the "
+                        + "query string to UTC."
+                ),
+                @MapParam.MapParamEntry(
+                    name = LOCALE_PARAM_NAME,
+                    type = "keyword",
+                    valueHint = { "standard" },
+                    description = "The locale to use when parsing the date, relevant when parsing month names or week days."
+                ) },
```

Comment:
I wonder if there's a way we can declare this thing once and reuse it. Java annotations have such weird rules.